etl/steps/data/garden/hmd/2022-11-04/life_tables.py

```diff
@@ -0,0 +1,97 @@
+import json
+from typing import List, cast
+
+import pandas as pd
+from owid.catalog import Dataset, Table
+from owid.catalog.utils import underscore_table
+from owid.datautils import geo
+from structlog import get_logger
+
+from etl.helpers import Names
+from etl.paths import DATA_DIR
+
+log = get_logger()
+
+# naming conventions
+N = Names(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("life_tables.start")
+
+    # read dataset from meadow
+    ds_meadow = Dataset(DATA_DIR / "meadow/hmd/2022-11-04/life_tables")
+
+    # init dataset
+    ds_garden = Dataset.create_empty(dest_dir)
+    ds_garden.metadata = ds_meadow.metadata
+    print(N.metadata_path)
+    ds_garden.metadata.update_from_yaml(N.metadata_path)
+
+    # build tables
+    tables_names = ds_meadow.table_names
+    for table_name in tables_names:
+        log.info(f"life_tables:{table_name}.start")
+        tb_garden = make_table(ds_meadow, table_name)
+        ds_garden.add(tb_garden)
+        ds_garden.save()
+        log.info(f"life_tables:{table_name}.end")
+
+    log.info("life_tables.end")
+
+
+def make_table(ds_meadow, table_name: str) -> Table:
+    log.info(f"Building table {table_name}...")
+    tb_meadow = ds_meadow[table_name]
+
+    # Country management
+    df = pd.DataFrame(tb_meadow)
+    df = clean_countries(df)
+
+    # Build table
+    tb_garden = underscore_table(Table(df))
+    tb_garden.metadata = tb_meadow.metadata
+
+    # Edit variables
+    for col in tb_garden.columns:
+        tb_garden[col].metadata = tb_meadow[col].metadata
+
+    # Edit table
+    tb_garden.update_metadata_from_yaml(N.metadata_path, table_name)
+    tb_garden = tb_garden.set_index(["country", "year", "age"])
```

Comment:
I'd suggest using `verify_integrity=True`, unless you know that there can be duplicates for some reason. Just as an easy additional sanity check.